src/sagemaker_xgboost_container/mms_patch/model_server.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+import os
+import signal
+import subprocess
+import sys
+
+import pkg_resources
+import psutil
+from retrying import retry
+
+import sagemaker_inference
+from sagemaker_inference import default_handler_service, environment, logging, utils
+from sagemaker_inference.environment import code_dir
+
+logger = logging.get_logger()
+
+DEFAULT_HANDLER_SERVICE = default_handler_service.__name__
+MMS_CONFIG_FILE = os.path.join('/etc', 'sagemaker-mms.properties')
+DEFAULT_MMS_CONFIG_FILE = pkg_resources.resource_filename(sagemaker_inference.__name__, '/etc/default-mms.properties')
+DEFAULT_MMS_LOG_FILE = pkg_resources.resource_filename(sagemaker_inference.__name__, '/etc/log4j.properties')
+DEFAULT_MMS_MODEL_DIRECTORY = os.path.join(os.getcwd(), '.sagemaker/mms/models')
+DEFAULT_MMS_MODEL_NAME = 'model'
+
+PYTHON_PATH_ENV = 'PYTHONPATH'
+REQUIREMENTS_PATH = os.path.join(code_dir, "requirements.txt")
+MMS_NAMESPACE = "com.amazonaws.ml.mms.ModelServer"
+
+
+def start_model_server(is_multi_model=False, handler_service=DEFAULT_HANDLER_SERVICE, config_file=None):
```

Comment:
Is this going to be merged in sagemaker-inference-toolkit? 

If so, this change will require us to change framework containers code (to set is_multi_model value). I would let sagemaker-inference-toolkit to get the SAGEMAKER_MULTI_MODEL env var value and determine which mms config to use so that mxnet/pytorch container does not have to make code changes.

Reply:
I'm really not a fan of using env vars; you don't know what you might be overriding, and you don't know when it will be overriden. If the value is not going to be used in any other context it should be passed as a method argument. This file should be useable as it has a default value for the method arguments.